src/converters.ts

```diff
@@ -184,25 +184,65 @@ const boolean = new Converter<boolean, boolean>({
 });
 
 export interface DecimalOptions {
-  maxWholeDigits?: number;
-  decimalPlaces?: number;
-  allowNegative?: boolean;
+  maxWholeDigits: number;
+  decimalPlaces: number;
+  allowNegative: boolean;
 }
 
-function decimal(options?: DecimalOptions) {
-  const maxWholeDigits: number =
-    options == null || !options.maxWholeDigits ? 10 : options.maxWholeDigits;
-  const decimalPlaces: number =
-    options == null || !options.decimalPlaces ? 2 : options.decimalPlaces;
-  const allowNegative: boolean =
-    options == null || options.allowNegative == null
-      ? true
-      : options.allowNegative;
+function decimal(
```

Comment:
It strikes me there's no reason for these functions to be nested. I would suggest we extract the decimal-related helper stuff (everything but the converter) into a 'decimal.ts' file.

Reply:
This includes the other decimal support functions we already wrote.